examples/window-app/common/src/WindowApp.cpp

```diff
@@ -38,12 +42,23 @@ void WindowApp::Timer::Timeout()
 
 void WindowApp::Button::Press()
 {
-    Instance().PostEvent(Button::Id::Up == mId ? WindowApp::Event::UpPressed : WindowApp::Event::DownPressed);
+    EventId event = Button::Id::Up == mId ? EventId::UpPressed : EventId::DownPressed;
+    Instance().PostEvent(WindowApp::Event(event));
 }
 
 void WindowApp::Button::Release()
 {
-    Instance().PostEvent(Button::Id::Up == mId ? WindowApp::Event::UpReleased : WindowApp::Event::DownReleased);
+    Instance().PostEvent(Button::Id::Up == mId ? EventId::UpReleased : EventId::DownReleased);
+}
+
+WindowApp::Cover & WindowApp::GetCover()
+{
+    return mCoverList[mCurrentCover];
+}
+
+WindowApp::Cover & WindowApp::GetCover(chip::EndpointId endpoint)
+{
+    return mCoverList[endpoint > 0 ? endpoint - 1 : 0];
```

Comment:
Should we check that endpoint index does not overfflow the  mCoverList size (and that it does not go negative either)?

A comment regarding valid ranges may help: why will 0 and 1 return the same value?

Reply:
I guess It is because  utilities clusters are using mainly endpoint 0 , so 0 must not be used by this cluster.
Here we expect always starting at endpoint = 1 !? right @ricardo-casallas  ?
But otherwise indeed how do you prevent overflow if endpoint is for example 1000 ?

Reply:
This implementation is assuming that index=endpoint-1. That may not be true for a general case, but this app is an example. Still, I don't like it. I have modified this function to search for the given endpoint number, and return NULL if not found. This will tell the caller if the endpoint is supported, and behave accordingly.